pkg/registry/apis/iam/legacy/team.go

```diff
@@ -14,6 +15,79 @@ import (
 	"github.com/grafana/grafana/pkg/storage/unified/sql/sqltemplate"
 )
 
+type GetTeamInternalIDQuery struct {
+	OrgID int64
+	UID   string
+}
+
+type GetTeamInternalIDResult struct {
+	ID int64
+}
+
+var sqlQueryTeamInternalIDTemplate = mustTemplate("team_internal_id.sql")
+
+func newGetTeamInternalID(sql *legacysql.LegacyDatabaseHelper, q *GetTeamInternalIDQuery) getTeamInternalIDQuery {
+	return getTeamInternalIDQuery{
+		SQLTemplate: sqltemplate.New(sql.DialectForDriver()),
+		TeamTable:   sql.Table("team"),
+		Query:       q,
+	}
+}
+
+type getTeamInternalIDQuery struct {
+	sqltemplate.SQLTemplate
+	TeamTable string
+	Query     *GetTeamInternalIDQuery
+}
+
+func (r getTeamInternalIDQuery) Validate() error { return nil }
+
+func (s *legacySQLStore) GetTeamInternalID(
+	ctx context.Context,
+	ns claims.NamespaceInfo,
+	query GetTeamInternalIDQuery,
+) (*GetTeamInternalIDResult, error) {
+	query.OrgID = ns.OrgID
+	if query.OrgID == 0 {
+		return nil, fmt.Errorf("expected non zero org id")
+	}
+
+	sql, err := s.sql(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	req := newGetTeamInternalID(sql, &query)
+	q, err := sqltemplate.Execute(sqlQueryTeamInternalIDTemplate, req)
+	if err != nil {
+		return nil, fmt.Errorf("execute template %q: %w", sqlQueryTeamInternalIDTemplate.Name(), err)
+	}
+
+	rows, err := sql.DB.GetSqlxSession().Query(ctx, q, req.GetArgs()...)
+	defer func() {
+		if rows != nil {
+			_ = rows.Close()
+		}
+	}()
+
+	if err != nil {
+		return nil, err
+	}
+
+	if !rows.Next() {
+		return nil, errors.New("team not found")
+	}
+
+	var id int64
+	if err := rows.Scan(&id); err != nil {
```

Comment:
I'm not sure how db connection implemented here, but do we always perform Scan(), even for the single item to find?

Reply:
Yeah inside here we are using sqlx to query but sadly our session interface does not expose "X" variants where you can use `StructScan` so we have to scan each column